src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
> What I hadn't been thinking of was the perspective of a plugin author trying to protect someone from running a plugin using "stable API level X" with an nvim that has "in-development API level X".

Yes, some plugins could choose to be cautious and only use the latest stable version, while other would prefer to just use code for the master nvim version it was latest updated for (still supporting an older stable version with less features). Depending on the kind of plugin, "use neovim master, file an issue if something breaks" might be a good enough policy.

> That being said, I don't think the "pre-release" indicator should be set unless/until we make an API change in a development cycle. When an API change is introduced, API level and "pre-release" are both bumped. When a release is made, "pre-release" is disabled.

Yes, that was what I tried to say in the first post proposing the scheme, but you put it in clearer words :)